snuba/clickhouse/columns.py

```diff
@@ -68,10 +69,31 @@ def flatten(self, name: str) -> Sequence[FlattenedColumn]:
         return [FlattenedColumn(None, name, self)]
 
 
-class Nullable(ColumnType):
+class ColumnTypeWithModifier(ABC, ColumnType):
```

Comment:
I think a modifier should become an attribute of a column, and not a type in its own, but that's an orthogonal refactoring.